horovod/runner/gloo_run.py

```diff
@@ -280,7 +292,7 @@ def gloo_run(settings, nics, env, server_ip, command):
     # Each thread will use ssh command to launch the job on each remote host. If an
     # error occurs in one thread, entire process will be terminated. Otherwise,
     # threads will keep running and ssh session.
-    exec_command = _exec_command_fn(settings)
+    exec_command = _exec_command_fn(settings, terminate_on_worker_failure=True)
```

Comment:
So didn't we say this is the only useful behaviour? Then we don't need to make this configurable from outside `_exec_command_fn`.

Reply:
Sorry, this is the only useful behavior _for non-elastic gloo_.  Elastic gloo uses the same `_exec_command_fn` and while I don't have experience running elastic gloo, I'm assuming that it's best not to change that codepath.

Reply:
Alright, I checked, the `_exec_command_fn` is used by elastic horovod as well.

You can use `if not settings.elastic:` in `_exec_command_fn` to determine if `terminate_all_event` should be used.

Then, on non-zero exit code we can do:
```
if exit_code != 0 and terminate_all_event:
  if not any(ev.is_set() for ev in all_events):
    print('Terminating remaining workers after failure of Process {idx}.'.format(idx=index))
  terminate_all_event.set()
```

Reply:
Sure, I can use `settings.elastic`.